docs/fundamentals/code-analysis/quality-rules/ca2264.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "CA2264: Do not pass a non-nullable value to 'ArgumentNullException.ThrowIfNull'"
+description: "Learn about code analysis rule CA2264 - Do not pass a non-nullable value to 'ArgumentNullException.ThrowIfNull'"
+ms.date: 02/25/2024
+f1_keywords:
+  - CA2264
+helpviewer_keywords:
+  - CA2264
+dev_langs:
+  - CSharp
+  - VB
+---
+# CA2264: Do not pass a non-nullable value to 'ArgumentNullException.ThrowIfNull'
+
+| Property                            | Value                                                                   |
+|-------------------------------------|-------------------------------------------------------------------------|
+| **Rule ID**                         | CA2264                                                                  |
+| **Title**                           | Do not pass a non-nullable value to 'ArgumentNullException.ThrowIfNull' |
+| **Category**                        | [Usage](usage-warnings.md)                                              |
+| **Fix is breaking or non-breaking** | Non-breaking                                                            |
+| **Enabled by default in .NET 9**    | As warning                                                              |
+
+## Cause
+
+When a value is passed to a call to `ArgumentNullException.ThrowIfNull` which is known to never be null, an exception is never thrown, making the statement a no-op.
+
+## Rule description
+
+`ArgumentNullException.ThrowIfNull` throws when the passed argument is `null`. Certain constructs like non-nullable structs (except for <xref:System.Nullable%601>), 'nameof()' and 'new' expressions are known to never be null, so `ArgumentNullException.ThrowIfNull` will never throw.
+
+In the case of a struct, since `ArgumentNullException.ThrowIfNull` accepts an `object?`, the struct is boxed, causing an additional performance penalty.
```

Comment:
```suggestion
In the case of a struct, since `ArgumentNullException.ThrowIfNull` accepts an `object?`, the struct is boxed, which causes an additional performance penalty.
```